### PR TITLE
Explicitly request a large number of results from BB grading API

### PR DESCRIPTION
**Pull request by marcospri**
Due to a bug in Blackboard (see: https://github.com/hypothesis/lms/pull/5839)
we fetch results for all students and just pick the right one.

We didn't implement pagination together with that change so we are
missing students that are part of the course but are present in other
pages.

For now will just request a large number of results and alert when we
still hit the pagination limit.

Also see: https://hypothes-is.slack.com/archives/C2C2U40LW/p1726613664446289

### Testing

- Login as `blackboardteacher` in https://aunltd-test.blackboard.com/
- Launch the [LTI 1.3 assignment test assignment](https://aunltd-test.blackboard.com/webapps/blackboard/content/contentWrapper.jsp?content_id=_2085_1&displayName=localhost+PDF+-+LTI1.3&course_id=_19_1&navItem=content&href=%2Fwebapps%2Fblackboard%2Fexecute%2Fblti%2FlaunchPlacement%3Fblti_placement_id%3D_148_1%26content_id%3D_2085_1%26course_id%3D_19_1)

- Pick "Blackboard student" from the dropdown, the score will get updated and the comment will be now in the text area.

- Make a diff like:

```diff
diff --git a/lms/services/lti_grading/_v13.py b/lms/services/lti_grading/_v13.py
index f89737b44..53a21a800 100644
--- a/lms/services/lti_grading/_v13.py
+++ b/lms/services/lti_grading/_v13.py
@@ -48,7 +48,7 @@ class LTI13GradingService(LTIGradingService):
             # Using the user filter in the request removes the comment field in the response's body.
             del params["user_id"]
             # We add an explicit limit parameter to fetch all results
-            params["limit"] = 1000
+            params["limit"] = 1
 
         try:
             response = self._ltia_service.request(
```

to force the pagination to kick in.

- Read the score of a student again, you'll see this in the logs:

```2024-09-18 14:59:41,756 ERROR [lms.services.lti_grading._v13:74][MainThread] Blackboard grading reading result, container with paginated results```

### CODE REVIEW

File: lms/services/lti_grading/_v13.py [hunk 1]
```diff
@@ -47,6 +47,8 @@ def read_result(self, grading_id) -> GradingResult:
             # There's currently a bug in Blackboard's LTIA implementation.
             # Using the user filter in the request removes the comment field in the response's body.
             del params["user_id"]
+            # We add an explicit limit parameter to fetch all results
+            params["limit"] = 1000
```
Comment (marcospri) [inline lms/services/lti_grading/_v13.py:51]: This is basically the bugfix, at least for now. We might hit the page limit again but we'll get a sentry notification about it.

We use the same 1000 limit for example in Canvas API calls.

File: lms/services/lti_grading/_v13.py [hunk 2]
```diff
@@ -67,6 +69,11 @@ def read_result(self, grading_id) -> GradingResult:
             # Due to the bug mentioned above we didn't use the API filtering by user for blackboard.
             # We'll filter ourselves here instead.
             results = [result for result in results if result["userId"] == grading_id]
+            if response.links.get("next"):
+                # We want to be notified if we are leaving results unfetched and likely start following these links
+                LOG.error(
+                    "Blackboard grading reading result, container with paginated results"
```
Comment (acelaya) [inline lms/services/lti_grading/_v13.py:75]: Maybe the error should include the assignment ID? Or is that included contextually when the error is reported?

**Review by acelaya — APPROVED** [review]
Looks good. Just had some suggestions around adding extra context in comments.

File: lms/services/lti_grading/_v13.py (hunk 2 above)
Comment (marcospri) [reply lms/services/lti_grading/_v13.py:75]: Yeah, sentry should show all local variables' values

File: lms/services/lti_grading/_v13.py [hunk 3]
```diff
@@ -47,6 +47,8 @@ def read_result(self, grading_id) -> GradingResult:
             # There's currently a bug in Blackboard's LTIA implementation.
             # Using the user filter in the request removes the comment field in the response's body.
             del params["user_id"]
+            # We add an explicit limit parameter to fetch all results
```
[the remark this answers is not in the document]
Comment (marcospri) [reply lms/services/lti_grading/_v13.py:50]: Good point, improved the comment here.

**Pull request merged**